src/vte.c

```diff
@@ -276,10 +276,10 @@ void vte_init(void)
 		gint i;
 		const gchar *sonames[] = {
 #if GTK_CHECK_VERSION(3, 0, 0)
-			"libvte-2.91.so", "libvte-2.91.so.0", "libvte-2.91.dylib",
-			"libvte2_90.so", "libvte2_90.so.9", "libvte2_90.dylib",
+			"libvte-2.91.so", "libvte-2.91.so.0", "libvte-2.91.0.dylib",
+			"libvte2_90.so", "libvte2_90.so.9", "libvte2_90.9.dylib",
 #else
-			"libvte.so", "libvte.so.4", "libvte.so.8", "libvte.so.9", "libvte.dylib",
+			"libvte.so", "libvte.so.4", "libvte.so.8", "libvte.so.9", "libvte.9.dylib",
```

Comment:
shouldn't the symlinked names be also listed, just listed last?

Reply:
Ah, yeah, I wanted to ask the same question but forgot about it :-).

Is there any danger that e.g. libvte.9.dylib will be missing but libvte.dylib present on some system? Similarly, has there ever been some problem with VTE loading which would require that Geany tries all the possibilities like "libvte.so", "libvte.so.4", "libvte.so.8", "libvte.so.9"?

Reply:
I don't really know, but e.g. the `libvte.so` symlink is installed by the `libvte-dev` package on Debian; a user with only the `libvte9` package will use the `libvte.so.9` name.
BTW, we might like inverting the order of the versioned names so the newest is taken first?

Anyway, I don't know OSX, but if the VTE library sticks to the *NIX kind of rules it'll install a versioned version anyway, so I'd say it's unlikely to have the versioned version missing.  But OTOH, if the versioned version is missing and the unversioned one is present it'd be a little silly not to use it.

All this said, the underlying problem is more subtle, and I guess would happen with any 2 libraries we load that share used symbols -- e.g. if in the future there was a newer version of the VTE library and the plugins got linked to this, the same problem would happen until we list the new SONAME here… leading to the conclusion that the best thing would be to keep the symlink instead of copying the file :)  But well.  I don't have a strong option/solution here, but would naively think listing the unversioned name as a fallback would make sense.